Meson build system

MaxMignolet opened the pull request:
This PR contains a new build system based on Meson. It improves the detection of libraries which (almost) fully automated . It also supports incremental compilation. I tested it on an intel mac, Ubuntu 22, and MareNostrum 5 with minimal issues. Feedback and testing on a wider range of machines/installations would be welcome.

Installation instruction have been added to `INSTALL.md`.

mjv500 commented:
Trying out now. Real time comments below:
- does not recognize -framework accelerate on MacOS on init, but then uses it correctly during build. confusing
- presumes openblas, I had to degrade to blas
- The position of the Meson section in the installation doc mixes it with some other sections which may not be appropriate. To be checked.
- build works and runs super quickly on my laptop. 
- in ./tests I had to add a link to the bin directory - having a separate meson build dir is good, but some paths are missing. 2 tests fail. One is thermal_conductivity_2023 which probably does not need to be run, but is in make_all_testfiles.sh but not the main build tree (at least my build did not make it)

MaxMignolet commented:
List of improvements: 
 - I added symlinks to the master bin directory instead of just inside build/bin.
 - I degraded openblas to simply blas (which should be provided by lapack). 
 - I added `thermal_conductivity_2023` to the build project.
For the tests, dump_dynamical_matrices fails only on my mac. It does not produce any output file for some reasons. On my ubuntu all the test succeed.

And for the documentation, I will improve it in another iteration.

mjv500 reviewed (approved):
Looks correct and functional now. Might add examples for other machines, MPI checks and other stuff. I don't know Meson enough to confirm these other options are present/easy/possible

mjv500 commented:
The compile and test phase on GitHub times out after 20 min. @MaxMignolet you can increase the timeout duration by hand in .github/workflows/ci_simple_compilation_and_test.yml
